styleguide/core/Input.hbs

```diff
@@ -22,8 +22,8 @@
         <div class="{{blockName}}"{{extraAttributes}}>
 
             {{#defineBlockBody}}
-                {{element "input"}}
                 {{element "label"}}
+                {{element "input"}}
```

Comment:
@rhino88, @trhoppe: Given at the variety of potential types [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) and the possible attributes which apply to specific types, what if we had a template per type? 
`DateInput.hbs` or possibly `DateInput extend="Input.hbs"`

This would allow you to order the `label` in front of the `input` (a better default, IMO) while  keeping the ability to write a selector that can style the label based on the input type:
`.DateInput-label`

Would an approach like this be more intuitive for FE/BE, or more complicated?